sklearn/pipeline.py

```diff
@@ -20,12 +21,70 @@
 from .externals import six
 from .utils import tosequence
 from .utils.metaestimators import if_delegate_has_method
-from .externals.six import iteritems
 
 __all__ = ['Pipeline', 'FeatureUnion']
 
 
-class Pipeline(BaseEstimator):
+class _BasePipeline(six.with_metaclass(ABCMeta, BaseEstimator)):
+    """Handles parameter management for classifiers composed of named steps.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    def _replace_step(self, steps_attr, name, new_val):
+        new_steps = getattr(self, steps_attr)[:]
+        for i, (step_name, _) in enumerate(new_steps):
+            if step_name == name:
+                new_steps[i] = (name, new_val)
+                break
+        else:
+            raise ValueError('Unknown step name: %r' % name)
```

Comment:
The only place where this method is called currently requires the step name to be known. I'll state so and remove the else clause.